### PR TITLE
Psi documentation

### PR DESCRIPTION
**Pull request by gverbock**

### CODE REVIEW

**Comment by gverbock** [comment]
Dear @solegalli I made some changes in the documentation by expending the docs and removing some part of the docstring. I am not sure if it goes in the by you expected direction so I would appreciate if you could have a first high level look at it and let me know what parts are still missing.
Thanks in advance.

**Comment by solegalli** [comment]
Made a few edits, pushed here: https://github.com/gverbock/feature_engine/pull/4

Hi @gverbock 

Thanks a lot for the additions to the docs. It is looking really great.

I think it was great that you included demos with the `_split_dataframe()` method, because that allowed you to find a few bugs in the class, right?. So well done.

Looking forward, `_split_dataframe()` is a hidden method, and it is then not accessible to the users, at least not so straightforwardly. For example, if you type the class name, and then press tab, it will not show up, so you specifically need to now, first that it exists and then type it manually, as a user of the class.

Because it is a hidden method, we do not show it in the documentation, because it is not intended for users to use. Only those who want to know more about the class, would go to the source code and dig deeper. But for the majority of users, we just need to show the parameters and attributes that they can use (because they are not hidden).

So, unless you think that exposing these dataframes is crucial, in which case we would have to change the class functionality, I would go ahead and explain what the class does, but do not expose the internal functionality in the documentation.

I added a series of #TODO throughout the DropHighPSIFeatures.rst file, with things I was thinking about for this part of the docs. As a user, I would be interested in understanding how the class detects distribution shifts. So instead of using random features, I would suggest to include a feature which distribution changes in each one of the use cases (ie changes with porfoltio, changes in time, changes with customer id, etc), and then, if possible, add a plot showing the distribution change of the feature that will be selected for dropping, and one plot with a feature that will actually not be dropped.

What do you think about that?

I'd appreciate your input. Thanks a lot!

**Comment by gverbock** [comment]
@solegalli  Thanks for the comments and the improvements to the text. I will have a look at it. I see the point of each time having a feature with a shift and showing it is excluded instead of showing the intermediate matrices. I'll work on it.

I am not familiar with adding pictures to the rst file. Do you have an example in mind I could copy?

**Comment by solegalli** [comment]
Hi @gverbock 

To add images, first you save a png preferably, alternatively jpeg in this folder. And then to add it just follow a sample code in line 18 in [this file](https://github.com/feature-engine/feature_engine/edit/psi_documentation/docs/user_guide/imputation/MeanMedianImputer.rst).

**Comment by gverbock** [comment]
@solegalli  I made some changes to the text and included pictures to illustrate the difference in population above and below a given threshold. 

I still have an issue with the tests (test_docs), it failed on line 68 of the drop_psi_feature.py file, complaining it is an undefined label. I removed it to see if there was any other issue and it appears to be the only one.

```
More details in the :ref: `User Guide <psi_selection>
```
As this line of code, I was hoping you could help me with solving it (I have no clue ;-)  )

**Comment by gverbock** [comment]
I am also looking at a nice panel dataset where we could show how the PSI work is a real case scenario. So far I haven' t find anything so if you have a suggestion, please let me know.

**Review by solegalli — COMMENTED** [review]
Hi @gverbock 

Thank you so much for such an amount of detail and creativity in designing the docs and examples. This is so good, so good... I have no words :)

Really good. I am impressed. Thank you so much.

I made a PR to your repo with some minor changes in wording here and there.

What shall we do next?

shall we merge this PR and add a jupyter notebook at a later stage? or were you planning to make it part of this PR? you mentioned you were looking for datasets.

There is a credit risk one in the UCI Machine learning repository:
https://archive.ics.uci.edu/ml/datasets/default+of+credit+card+clients

Let me know!

Thanks again

**Comment by solegalli** [comment]
> @solegalli I made some changes to the text and included pictures to illustrate the difference in population above and below a given threshold.
> 
> I still have an issue with the tests (test_docs), it failed on line 68 of the drop_psi_feature.py file, complaining it is an undefined label. I removed it to see if there was any other issue and it appears to be the only one.
> 
> ```
> More details in the :ref: `User Guide <psi_selection>`.
> ```
> 
> As you wrote this line of code, I was hoping you could help me with solving it (I have no clue ;-) )

I don't get this error. Don't worry too much. If the error does not jump in circleci we are good to merge, and if it appears later, I will handle it, I hope :p

**Comment by gverbock** [comment]
@solegalli Thanks for the PR and the comments. I will have a look at it. My preference goes to merging without the Notebook and add it in a later PR. I need some time to find an appropriate data set and make a nice notebook.

File: docs/user_guide/selection/DropHighPSIFeatures.rst [hunk 1]
```diff
@@ -1,134 +1,858 @@
...
+    'var_4': 0.04492720670343145,
+    'var_5': 0.044886435640028144,
+    'group_means': 6.601444547497699,
+    'shifted_features': 0.3683642099948127}
+
+
+Here again, the object will remove the *group_means* and the *shifted_feature* columns
+from the dataframe.
+Also it is worth noticing that the PSI values of the `trans` object differ from
+those of the `transform` object with exception of the *group_means* variable.
```
[the remark this answers is not in the document]
Comment (gverbock) [reply docs/user_guide/selection/DropHighPSIFeatures.rst:696]: I will remove this as it is confusing. Indeed, the cut-off is different and therefore the PSI values are too,

File: docs/user_guide/selection/DropHighPSIFeatures.rst [hunk 2]
```diff
@@ -1,134 +1,858 @@
...
+
+Note that the data split is not done at random, but instead guided by the values in the reference
+variable indicated in `split_col`. Under the hood, the reference variable indicated in `split_col`
+is ordered, and the percentage of observations is determined with NumPy quantile. This means
+that the observations with smaller values in `split_col` will land in the reference dataset,
+and those with bigger values will go to the test set.
+
+If the rows in your dataset are sorted in time, this could be a good default option to split the
+dataframe in 2 and compute the PSI. This will for example be the case if your data set is made
+of several time series.
```
[the remark this answers is not in the document]
Comment (gverbock) [reply docs/user_guide/selection/DropHighPSIFeatures.rst:128]: I will rephrase this.

File: docs/user_guide/selection/DropHighPSIFeatures.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (gverbock) [reply docs/user_guide/selection/DropHighPSIFeatures.rst:696]: I also removed the part on the more detailed information at the end of the file. I will work on that in a later PR.

**Comment by solegalli** [comment]
Sounds good. Shall I merge then? Just to confirm.

**Comment by gverbock** [comment]
@solegalli Yes, you can merge!

**Pull request merged**